### PR TITLE
Fix direct react state mutation

### PR DESCRIPTION
**Pull request by keithjgrant**
## Summary

Fixes React state update that mutated the state array directly

## Type of Change

- [x] Bug fix
- [ ] Enhancement
- [ ] Tests
- [ ] Documentation
- [ ] Other (please specify)

## Risk Analysis - REQUIRED

- [ ] **High** — Broad platform impact (e.g., SWR config, shared framework components, authentication, routing, API wrappers, build/bundler config). Changes affect multiple workspaces or could cause widespread regressions.
- [ ] **Medium** — Scoped but cross-cutting (e.g., shared utility functions, changes to multiple pages within one workspace, component library updates, test infrastructure). Limited blast radius but touches common code paths.
- [x] **Low** — Narrowly scoped (e.g., single page fix, styling tweak, documentation, test-only changes, copy/string updates). Minimal risk of unintended side effects.

### CODE REVIEW

**Comment by keithjgrant** [comment]
This was a one-line change, but that line was not covered under test. (SonarQube was fine with this, but CodeCov didn't like it). I restructured the function to be testable. I don't think I can get more coverage without completely restructuring the main hook.

File: frontend/awx/views/jobs/JobOutput/useJobOutput.test.ts [hunk 1]
```diff
@@ -37,6 +38,47 @@ const dateRangeFilter = {
...
+
+    expect(result[1]).toBe(newEvent);
+  });
+
+  it('should return a shallow copy when newEvents is empty', () => {
+    const existing: Record<number, JobEvent> = { 1: makeEvent(1) };
+    const result = applyBatchedEvents(existing, []);
+
+    expect(result).not.toBe(existing);
+    expect(result).toEqual(existing);
```
Comment (keithjgrant) [inline frontend/awx/views/jobs/JobOutput/useJobOutput.test.ts:78]: **Nit** (0 points): Consider adding a test for the empty initial state path, since that is the production initial value from `useState({})`. The behavior is trivially correct via spread, but it would make coverage comprehensive.

```suggestion
  it('should return a shallow copy when newEvents is empty', () => {
    const existing: Record<number, JobEvent> = { 1: makeEvent(1) };
    const result = applyBatchedEvents(existing, []);

    expect(result).not.toBe(existing);
    expect(result).toEqual(existing);
  });

  it('should add events to an empty record', () => {
    const result = applyBatchedEvents({}, [makeEvent(1), makeEvent(2)]);

    expect(result[1]).toEqual(makeEvent(1));
    expect(result[2]).toEqual(makeEvent(2));
  });
```

**Review by keithjgrant — COMMENTED** [review]
## Code Review: Fix direct react state mutation (#3360)

**Verdict**: READY_FOR_HUMAN_REVIEW
**Score**: 10.0/10 (Functionality 10, Security 10, Quality 10)

No critical or major findings. The fix correctly addresses a React state mutation bug where `setJobEvents` was mutating state in-place and returning the same reference, which could cause React to skip re-renders. The extracted `applyBatchedEvents` function creates a new object via spread, and tests verify non-mutation, merge, overwrite, and empty-input behavior.

One optional nit: add a test for the empty initial state path (`applyBatchedEvents({}, [...])`) to match the production `useState({})` starting value. See inline comment.

*Generated with AI assistance: Claude Code / Opus 4.6 (Anthropic)*

**Review by vidyanambiar — APPROVED** [review]
